### DISCUSSION TITLE
Not found error

### DISCUSSION THREAD

**Discussion by doriansgithub** [sbooth/SFBAudioEngine#247] [General]
I've installed the framework and can't add the player to my main player.

SFBAudioPlayer.h' file not found

**Comment by sbooth** [comment]
Are you using Swift or Objective-C? Is `<SFBAudioEngine/SFBAudioEngine.h>` included in your bridging header or in your main player?

**Comment by doriansgithub** [comment]
Hi, and thank you for your time.

The project uses both ObjC and Swift, but the main player is ObjC.

If I try to import it in the main player, I get errors.



If I add any of these to my bridge file, I get an error.

#include <SFBAudioEngine/SFBAudioEngine.h>
#import "SFBAudioEngine/SFBAudioEngine.h”

**Comment by doriansgithub** [comment]
This is how I imported the framework.

**Reply by sbooth** [reply]
Can you post the specific errors you're getting?

**Comment by doriansgithub** [comment]


**Reply by doriansgithub** [reply]
Hold your horses, it looks like I figured it out.  Deleted the references to the frameworks from my project and left only in the SFBAudioEngine side.

**Reply by doriansgithub** [reply]
I got it to run, but it won't play.

![image](https://user-images.githubusercontent.com/5715600/205338482-0dacd019-39c0-43fb-9bb9-786e718a4c2b.png)

Do I have to implemente something similar to this?
	init(url: URL) {
		self.url = url
		if let audioFile = try? AudioFile(readingPropertiesAndMetadataFrom: url) {
			self.properties = audioFile.properties
			self.metadata = audioFile.metadata
		} else {
			self.properties = AudioProperties()
			self.metadata = AudioMetadata()
		}
	}

Thanks!

**Reply by sbooth** [reply]
It looks like that URL is an item in the user's music library. Since it isn't a file URL it won't work correctly with `SFBAudioEngine`. I am not sure if it's possible to play content from the user's music library (at least on iOS) without using the Media Player framework or AVFoundation. It might be possible to jump through some hoops and create a wrapper around `AVAssetExportSession`.

**Comment by doriansgithub** [comment]
Ok, so if it’s an DSD song, it would play?Sent from my iPhoneOn Dec 2, 2022, at 12:08 PM, Stephen Booth ***@***.***> wrote:﻿
It looks like that URL is an item in the user's music library. Since it isn't a file URL it won't work correctly with SFBAudioEngine. I am not sure if it's possible to play content from the user's music library (at least on iOS) without using the Media Player framework or AVFoundation. It might be possible to jump through some hoops and create a wrapper around AVAssetExportSession.

**Reply by sbooth** [reply]
If it's a DSD64 file it should be playable.

**Comment by doriansgithub** [comment]
And how would an EQ work? Is there a sample?

**Reply by sbooth** [reply]
See #245 . There currently isn't a sample.
